src/averageAges.js

```diff
@@ -55,7 +74,37 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let counter = 0;
+
+  if (onlyWithSon) {
+    const ageDifference = people.map(person => {
+      const currentMother = people.filter(mother =>
+        person.mother === mother.name && person.sex === 'm');
+
+      if (currentMother.length > 0) {
+        counter++;
+      }
+
+      return currentMother.length > 0 ? person.born - currentMother[0].born : 0;
+    });
+    const sumOfAges = ageDifference.reduce((sum, num) => sum + num, 0);
+
+    return sumOfAges / counter;
+  } else {
+    const ageDifference = people.map(person => {
+      const currentMother = people.filter(mother =>
+        person.mother === mother.name);
+
+      if (currentMother.length > 0) {
+        counter++;
+      }
+
+      return currentMother.length > 0 ? person.born - currentMother[0].born : 0;
+    });
+    const sumOfAges = ageDifference.reduce((sum, num) => sum + num, 0);
+
+    return sumOfAges / counter;
+  }
```

Comment:
same. avoid code repeating. `onlyWithSon` value effect only the sample you should work with